docs/_developers/devguide.md

```diff
@@ -1,39 +1,70 @@
+GitHub is used to track issues and contribute code, suggestions, feature requests or
+documentation.
+
+If you'd like to contribute to go-ethereum, please fork, fix, commit and send a pull
+request (PR) for the maintainers to review and merge into the main code base. If you wish
+to submit more complex changes though, please check up with the core devs first on [our
+gitter channel](https://gitter.im/ethereum/go-ethereum) to ensure those changes are in
+line with the general philosophy of the project and/or get some early feedback which can
+make both your efforts much lighter as well as our review and merge procedures quick and
+simple.
+
+PRs need to be based on and opened against the `master` branch (unless by explicit
+agreement, you contribute to a complex feature branch).
+
+Your PR will be reviewed according to the [Code Review guidelines](../developers/code-review-guidelines).
+
+We encourage a PR early approach, meaning you create the PR the earliest even without the
+fix/feature. This will let core devs and other volunteers know you picked up an issue.
+These early PRs should indicate 'in progress' status.
+
+## Building and Testing
+
+We assume that you have [Go installed](https://golang.org/doc/install), and `GOPATH` is set.
 
 **Note**:You must have your working copy under `$GOPATH/src/github.com/ethereum/go-ethereum`.
 
-Since `go` does not use relative path for import, working in any other directory will have no effect, since the import paths will be appended to `$GOPATH/src`, and if the lib does not exist, the version at master HEAD will be downloaded.
+Since Go does not use relative path for import, working in any other directory will have
+no effect, since the import paths will be appended to `$GOPATH/src`, and if the lib does
+not exist, the version at master HEAD will be downloaded.
 
-Most likely you will be working from your fork of `go-ethereum`, let's say from `github.com/nirname/go-ethereum`. Clone or move your fork into the right place:
+Most likely you will be working from your fork of `go-ethereum`, let's say from
+`github.com/nirname/go-ethereum`. Clone or move your fork into the right place:
 
 ```
 git clone git@github.com:nirname/go-ethereum.git $GOPATH/src/github.com/ethereum/go-ethereum
 ```
 
 ### Managing Vendored Dependencies
 
-All other dependencies are tracked in the vendor/ directory. We use [govendor](https://github.com/kardianos/govendor) to manage them.
+All other dependencies are tracked in the vendor/ directory. We use
+[govendor](https://github.com/kardianos/govendor) to manage them.
 
-If you want to add a new dependency, run `govendor fetch <import-path>`, then commit the result.
+If you want to add a new dependency, run `govendor fetch <import-path>`, then commit the
+result.
 
-If you want to update all dependencies to their latest upstream version, run `govendor fetch +v`.
+If you want to update all dependencies to their latest upstream version, run `govendor
```

Comment:
It's a matter of taste, but I would break the line before the command block, not within